opentech/apply/funds/views.py

```diff
@@ -493,15 +493,18 @@ def form_valid(self, form):
                 related=revision,
             )
 
-        action = set(self.request.POST.keys()) & set(self.transitions.keys())
-        transition = self.transitions[action.pop()]
-
-        self.object.perform_transition(
-            transition.target,
-            self.request.user,
-            request=self.request,
-            notify=not (revision or submitting_proposal),  # Use the other notification
-        )
+        try:
+            action = set(self.request.POST.keys()) & set(self.transitions.keys())
```

Comment:
Keep as little as possible inside the try except. this first line should be outside.